aea/protocols/generator/base.py

```diff
@@ -1900,6 +1900,9 @@ def _protocol_yaml_str(self) -> str:
         protocol_yaml_str = "name: {}\n".format(self.protocol_specification.name)
         protocol_yaml_str += "author: {}\n".format(self.protocol_specification.author)
         protocol_yaml_str += "version: {}\n".format(self.protocol_specification.version)
+        protocol_yaml_str += "type: {}\n".format(
```

Comment:
why do yaml  generated as string not using yaml dump? to preserve order of the keys?